src/meshpy/four_c/locsys_condition.py

```diff
@@ -34,25 +37,52 @@ class LocSysCondition(BoundaryCondition):
 
     def __init__(
         self,
-        geometry_set,
-        rotation,
+        geometry_set: GeometrySet,
+        rotation: Rotation,
+        *,
+        function_array=None,
+        update_node_position: bool = False,
+        use_consistent_node_normal: bool = False,
         **kwargs,
     ):
         """Initialize the object.
 
         Args
         ----
-        geometry_set: GeometrySet
-            Geometry that this boundary condition acts on.
-        rotation: Rotation
-            Object that represents the rotation of the coordinate system.
+        geometry_set: Geometry that this boundary condition acts on.
+        rotation: Object that represents the rotation of the coordinate system.
+        function_array: (list[Function])
+            list containing functions
+        update_node_position: Flag to enable the updated node position
+        use_consistent_node_normal: Flag to use a consistent node normal
         """
 
+        # Validate provided function array.
+        if function_array is not None:
+            ensure_length_of_function_array(function_array, 3)
+        else:
+            function_array = [0, 0, 0]
+
+        condition_string = (
+            "ROTANGLE {} {} {} ".format(*rotation.get_rotation_vector())
+            + "FUNCT {} {} {} "
+            + f"USEUPDATEDNODEPOS {int(update_node_position)}"
+        )
+
+        # Append the condition string with consistent normal type for line and surface geometry
+        if (
+            geometry_set.geometry_type is mpy.geo.line
+            or geometry_set.geometry_type is mpy.geo.surface
+        ):
+            condition_string = (
+                condition_string
+                + f" USECONSISTENTNODENORMAL {int(use_consistent_node_normal)}"
+            )
+
```

Comment:
Maybe we can add this check here
```suggestion
elif use_consistent_node_normal:
    raise ValueError("The keyword use_consistent_node_normal only works for line and surface geometries")
```